pandas/tseries/frequencies.py

```diff
@@ -405,11 +427,19 @@ def _get_wom_rule(self) -> str | None:
         #     if not lib.ismember(wdiffs, set([4, 5, -47, -49, -48])).all():
         #         return None
 
-        weekdays = unique(self.index.weekday)
+        # error: Item "DatetimeIndex" of "Union[DatetimeIndex, TimedeltaIndex]" has no
+        # attribute "weekday"
+        # error: Item "TimedeltaIndex" of "Union[DatetimeIndex, TimedeltaIndex]" has no
+        # attribute "weekday"
+        weekdays = unique(self.index.weekday)  # type: ignore[union-attr]
         if len(weekdays) > 1:
             return None
 
-        week_of_months = unique((self.index.day - 1) // 7)
+        # error: Item "DatetimeIndex" of "Union[DatetimeIndex, TimedeltaIndex]" has no
+        # attribute "day"
+        # error: Item "TimedeltaIndex" of "Union[DatetimeIndex, TimedeltaIndex]" has no
+        # attribute "day"
+        week_of_months = unique((self.index.day - 1) // 7)  # type: ignore[union-attr]
```

Comment:
doesnt need to be this PR, but looks like the solution here is to pin down self.index as DatetimeIndex here